plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/PomXmlValidator.java

```diff
@@ -16,35 +16,49 @@
 
 package com.google.cloud.tools.eclipse.appengine.validation;
 
-import java.io.IOException;
-import java.util.Map;
+import java.util.ArrayList;
 
-import javax.xml.parsers.ParserConfigurationException;
+import org.eclipse.core.resources.IResource;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
 
-import org.eclipse.core.resources.IFile;
-import org.eclipse.core.runtime.CoreException;
-import org.xml.sax.SAXException;
+public class PomXmlValidator implements XmlValidationHelper {
 
-public class PomXmlValidator extends AbstractXmlValidator {
-
-  /**
-   * Clears all problem markers from the resource, then adds a Maven plugin marker
-   * to pom.xml if the App Engine Maven plugin is obsolete.
-   */
-  @Override
-  protected void validate(IFile resource, byte[] bytes)
-      throws CoreException, IOException, ParserConfigurationException {
-    try {
-      deleteMarkers(resource);
-      SaxParserResults parserResults = PomParser.readXml(bytes);
-      Map<BannedElement, Integer> bannedElementOffsetMap =
-          ValidationUtils.getOffsetMap(bytes, parserResults);
-      for (Map.Entry<BannedElement, Integer> entry : bannedElementOffsetMap.entrySet()) {
-        createMarker(resource, entry.getKey(), entry.getValue());
+  public ArrayList<BannedElement> checkForElements(IResource resource, Document document) {
+    DocumentLocation location = null;
```

Comment:
location should be declared inside the for loop to avoid accidental reuse of values from previous iterations